static/ast-view.js

```diff
@@ -141,6 +145,12 @@ Ast.prototype.onCompilerClose = function (id) {
     }
 };
 
+Ast.prototype.onLanguageChange = function (editorId, newLangId) {
+    if (editorId == this._editorid) {
+        monaco.editor.setLanguageModel(this.astEditor.getModel(), languages[newLangId].monaco);
```

Comment:
I don't know why this doesn't work here... somehow setLanguageModel doesn't update the language.

Reply:
(Note that the tests fail because automatic type coercion is disallowed (Js may convert operands type under the hood to make them comparable). You need to change the `==` to `===` which will disallow said type coercion)
If `languages` is set properly, (and of course the function is being hit at all), I can't see how it would not work either. I'll look at it further depending on what we come up on the other review message

Reply:
I added the test because I thought it's needed for some reason, but without it, it still doesn't work. The function _is_ hit (tested by changing the title of the AST pane, which does work), but yeah, somehow doesn't work. (Also doesn't work when hardcoding the language to change the pane to).

Reply:
Note that the test IS necessary, as otherwise changing the language on another (unreleated) editor would change the language here

Reply:
I know why this wasn't working.. You were calling `setLanguageModel` but the correct function is `setModelLanguage`. Welcome to the land of JS! :)

Reply:
oh man..... !